src/Umbraco.Web/Routing/DomainUtilities.cs

```diff
@@ -345,14 +331,17 @@ internal static Domain FindDomainInPath(IEnumerable<Domain> domains, string path
         /// <returns>The deepest wildcard Domain in the path, or null.</returns>
         /// <remarks>Looks _under_ rootNodeId but not _at_ rootNodeId.</remarks>
         internal static Domain FindWildcardDomainInPath(IEnumerable<Domain> domains, string path, int? rootNodeId)
+            => FindDomainInPath(domains, path, rootNodeId, true);
+
+        private static Domain FindDomainInPath(IEnumerable<Domain> domains, string path, int? rootNodeId, bool isWildcard)
         {
             var stopNodeId = rootNodeId ?? -1;
 
             return path.Split(',')
                        .Reverse()
                        .Select(int.Parse)
                        .TakeWhile(id => id != stopNodeId)
-                       .Select(id => domains.FirstOrDefault(d => d.ContentId == id && d.IsWildcard))
+                       .Select(id => domains.FirstOrDefault(d => d.ContentId == id && d.IsWildcard == isWildcard))
                        .FirstOrDefault(domain => domain != null);
         }
```

Comment:
I've created this private method to remove the duplicated code in `FindDomainInPath()` and `FindWildcardDomainInPath()`.